core-concepts/overview.mdx

```diff
@@ -0,0 +1,127 @@
+---
+title: Core Concepts Overview
+description: 'Understanding how Pipecat works: frames, processors, pipelines, and transport.'
+---
+
+Pipecat uses a pipeline-based architecture to handle real-time AI processing. Let's look at how this works in practice, then break down the key components.
+
+## Real-time Processing in Action
+
+Consider how a voice assistant processes a user's question and generates a response:
+
+<Frame>![Real-time processing pipeline](/images/architecture-1.png)</Frame>
+
+Instead of waiting for complete responses at each step, Pipecat processes data in small units called frames that flow through the pipeline:
+
+1. Speech is transcribed in real-time as the user speaks
+2. Transcription is sent to the LLM as it becomes available
+3. LLM responses are processed as they stream in
+4. Text-to-speech begins generating audio for early sentences while later ones are still being generated
+5. Audio playback starts as soon as the first sentence is ready
+6. LLM context is aggregated and updated continuously and in real-time
+
+This streaming approach creates natural, responsive interactions.
+
+## Architecture Overview
+
+Here's how Pipecat organizes these processes:
+
+<Frame>![Platform architecture](/images/architecture-2.png)</Frame>
```

Comment:
re: this image- should we rename "AudioFrame" and "LLMContextFrame" as those are no longer frame types?

Reply:
I'm using InputAudioRawFrame for audio and I updated `LLMContextFrame` to `OpenAILLMContextFrame`, which is more accurate.